src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -404,13 +416,40 @@ static String validateConfig(Map<String, String> config)
     return connectorName;
   }
 
+  /**
+   * modify invalid application name in config and return the generated application name
+   * @param config input config object
+   * @return valid application name
+   */
+  public static String convertAppName(Map<String, String> config)
+  {
+    String appName =
+      config.getOrDefault(SnowflakeSinkConnectorConfig.NAME, "");
+    // If appName is empty the following call will throw error
+    String validAppName = generateValidName(appName, new HashMap<String, String>());
+
+    config.put(SnowflakeSinkConnectorConfig.NAME, validAppName);
```

Comment:
If config map was updated, do we need return app name again?

Reply:
ack